CMakeLists.txt

```diff
@@ -0,0 +1,336 @@
+# minimum required vertion of cmake
+cmake_minimum_required (VERSION 3.2)
+
+# project name
+project (raqm)
+
+##############################3
+
+#enable compiler warning for GNU C
+if(CMAKE_COMPILER_IS_GNUCC)
+add_definitions (-Wall  -Wno-portability)
+#enable C99 mode
+add_definitions(-std=gnu99)
+endif()
+
+#enable compiler warning for MSVC
+if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  set(CMAKE_C_FLAGS "/W4 /WX /wd4068")
+endif()
+
+#not working yet.. check
+#if(WIN32 AND NOT(MSYS OR CYGWIN OR MINGW) )
+#set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W4 /wd4068")
+#endif()
+
+
+
+#set the values of raqam.pc file
+set (prefix /usr/local)
+set (exec_prefix ................................)
+set (libdir /lib)
+set (includedir /include)

```

Comment:
All these hard-coded values are just ugly, with autotools they are derived from the configure script. I don’t like this hard-coding since the user can install Raqm under a different prefix and then these values will be wrong. They also definitely wrong already; our libdir is `$PREFIX/lib` not just `/lib`.